### PR TITLE
CR-1085 - 3 endpoints disallowed if survey not CENSUS

### PR DESCRIPTION
**Pull request by devenrawat**
# Motivation and Context

During CCS Census and CCS overlap. There are 3 endpoints where check for surveyType attribute on the case was required.
If surveyType is not CENSUS then Response will be BAD_REQUEST with message "Operation not permissible for a CCS Case"
1.  POST /cases
2.  POST /cases/ {caseId}/fulfilment/post 
3.  POST /cases/{caseId}/fulfilment/sms

# What has changed
Added condition at two places in service
1.   `modifyCase` method to check surveyType from RM
2. `createFulfilmentRequestPayload` same check as above.
One more change is done to make surveyType attribute of all Cases  from cache/firestore  as CENSUS in method `getCaseFromRmOrCache`

Three new tests are written for all three endpoints and json file is changed to make sure the surveyType is CENSUS. 

# Links
https://collaborate2.ons.gov.uk/jira/browse/CR-1085

### CODE REVIEW

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java [hunk 1]
```diff
@@ -95,10 +95,12 @@
       "All Northern Ireland calls from CE Managers are to be escalated to the NI management team.";
   private static final String UNIT_LAUNCH_ERR_MSG =
       "A CE Manager form can only be launched against an establishment address not a UNIT.";
+  public static final String CCS_CASE_ERROR_MSG = "Operation not permissible for a CCS Case";
```
Comment (rpcatling) [inline src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java:98]: this should be private

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java [hunk 2]
```diff
@@ -95,10 +95,12 @@
...
   private static final List<DeliveryChannel> ALL_DELIVERY_CHANNELS =
       List.of(DeliveryChannel.POST, DeliveryChannel.SMS);
 
   private static final String SCOTLAND_COUNTRY_CODE = "S";
+  public static final String CENSUS = "CENSUS";
```
Comment (rpcatling) [inline src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java:103]: I think we should delete this CENSUS variable, and use **appConfig.getSurveyName()** instead.

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java [hunk 3]
```diff
@@ -498,6 +500,9 @@ public CaseDTO modifyCase(ModifyCaseRequestDTO modifyRequestDTO) throws CTPExcep
     UUID caseId = originalCaseId;
 
     CaseContainerDTO caseDetails = getCaseFromRmOrCache(originalCaseId, true);
+    if (!CENSUS.equalsIgnoreCase(caseDetails.getSurveyType())) {
```
Comment (rpcatling) [inline src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java:503]: potential for this to be factored into a resusable method  **validateSurveyType()**  and use appConfig.getSurveyName().

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java [hunk 4]
```diff
@@ -856,7 +861,9 @@ private FulfilmentRequest createFulfilmentRequestPayload(
         .debug("Entering createFulfilmentEvent method in class CaseServiceImpl");
 
     CaseContainerDTO caze = getCaseFromRmOrCache(caseId, false);
-
+    if (!CENSUS.equalsIgnoreCase(caze.getSurveyType())) {
```
Comment (rpcatling) [inline src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java:864]: could call method **validateSurveyType**

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java [hunk 5]
```diff
@@ -944,6 +951,7 @@ private CaseContainerDTO getCaseFromRmOrCache(UUID caseId, boolean getCaseEvents
         if (cachedCase.isPresent()) {
           log.with("caseId", caseId).info("Using stored case details");
           caze = caseDTOMapper.map(cachedCase.get(), CaseContainerDTO.class);
+          caze.setSurveyType(CENSUS);
```
Comment (rpcatling) [inline src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java:954]: use appConfig.getSurveyName()

File: src/test/resources/uk/gov/ons/ctp/integration/contactcentresvc/CCSvcBeanMapperTest.CaseContainerDTO.json [hunk 6]
```diff
@@ -45,5 +45,5 @@
     }
   ],
   "collectionExerciseId": "22684ede-7d5f-4f53-9069-2398055c61b2",
-  "surveyType": "x"
+  "surveyType": "CENSUS"
 }
```
Comment (rpcatling) [inline src/test/resources/uk/gov/ons/ctp/integration/contactcentresvc/CCSvcBeanMapperTest.CaseContainerDTO.json:49]: just while you are here, please add a new line at the end to get of the github warning about no end-of-line, thanks.

**Review by rpcatling — CHANGES_REQUESTED** [review]
Functionally good, just some suggested tidy ups, and make use of existing application config surveyType.

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java (hunk 1 above)
Comment (devenrawat) [reply src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java]: This is changed to private: Done

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java (hunk 2 above)
Comment (devenrawat) [reply src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java]: This constant removed and used appConfig.getSurveyName()

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java (hunk 3 above)
Comment (devenrawat) [reply src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java]: Done: method added

File: src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java (hunk 4 above)
Comment (devenrawat) [reply src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java]: Done

File: src/test/resources/uk/gov/ons/ctp/integration/contactcentresvc/CCSvcBeanMapperTest.CaseContainerDTO.json (hunk 6 above)
Comment (devenrawat) [reply src/test/resources/uk/gov/ons/ctp/integration/contactcentresvc/CCSvcBeanMapperTest.CaseContainerDTO.json]: Added new line at the end

**Review by rpcatling — APPROVED** [review]

**Review by philwhiles — APPROVED** [review]

**Pull request merged**